airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -115,10 +116,11 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
 
             if config.get("account_type", "Seller") == "Seller":
                 stream_to_check = Orders(**stream_kwargs)
+                next(stream_to_check.read_records(sync_mode=SyncMode.full_refresh))
             else:
-                stream_to_check = VendorSalesReports(**stream_kwargs)
-
-            next(stream_to_check.read_records(sync_mode=SyncMode.full_refresh))
+                stream_to_check = VendorOrders(**stream_kwargs)
```

Comment:
Should we add a notice in the documentation? (e.g. "If you want to pass the check with a vendor or sales account, you'll need to have access to certain APIs.")

Reply:
Fixed